### PR TITLE
Added CONSENT check for the insertion of DICSS

### PR DESCRIPTION
**Pull request by The-Feminist-Software-Foundation**

### CODE REVIEW

**Comment by letsgetrandy** [comment]
Appreciated, but I feel that at this point, consent is too intrusive.  It should be given in advance.

Maybe something more like:

```
DICSS.consent = true;
```

and later, 

```
if (DICSS.consent) {
   // ...
}
```

**Comment by The-Feminist-Software-Foundation** [comment]
I'm sorry, but do you have a degree in Women's Studies?  Because I have, and I can assuredly tell you, that consent is not something that can be given in advance.

Consent is not a contract.  Consent can be given and withdrawn at any moment if the woman chooses so.  To presume otherwise is basically rape.

You don't want to be labelled as a rape-apologist, do you?  By coding in rapey style, you're enabling and perpetuating the rape culture that we all live in in our daily lives.  You don't want to be responsible for the rape of the millions of women every day, do you?

**Comment by letsgetrandy** [comment]
@The-Feminist-Software-Foundation I seriously love you.

**Comment by Zaskoda** [comment]
@letsgetrandy What about a hybrid. Start with soft consent check early on to validate if the connection is appropriate. Knowing that some stacks of code can be unstable, keep pre-insertion consent verification that TFSF pull requested in place. However, add a check that if the consent flag is flipped off before insertion, the caller is released and the caller's name is tracked in a queue so that future instances of DICSS don't lose resouces initializing for the same caller. Everybody wins.

**Comment by stal888** [comment]
:+1: I am tired of random javascript executing without my active consent.

File: dicss.js [hunk 1]
```diff
@@ -72,7 +72,12 @@
 	window.DICSS = {
 		"putIn": function(selector, properties) {
 			if (properties !== void 0) {
-				addRule.apply(this, arguments);
+				var consent = confirm("Are you sure you consent to the insertion of DICSS?");
+				if (consent == true) {
```
Comment (Kagami) [inline dicss.js:76]: Shouldn't it be `===`? [Beware of `==`](https://dorey.github.io/JavaScript-Equality-Table/).
Not required here but better to hold good coding standarts.

**Comment by Kyokou** [comment]
>Consent is not a contract. Consent can be given and withdrawn at any moment if the woman chooses so. To presume otherwise is basically rape.

So in the middle of intercourse, if a women withdraws her consent, how long do I have to exit before it's considered rape? Or is it immediately rape?

**Comment by miguelcobain** [comment]
@The-Feminist-Software-Foundation observations are right, but _only if_ we assume that DICSS is only inserted into women.
Especially in production environments, use cases may vary a lot, including scenarios where a woman isn't needed at all, and thus, no consent is required.

Would love to see this PR reflect more generic scenarios.

**Comment by WaveringAna** [comment]
What if I'm not putting DICSS in .person? Do I still need consent?

File: dicss.js [hunk 2]
```diff
@@ -72,7 +72,12 @@
...
+					addRule.apply(this, arguments);
+				} else {
+					alert("CONSENT VIOLATED!");
```
Comment (Joimer) [inline dicss.js:79]: While `alert` is a good way to display a yell, which might be adequate in this situation, it is highly unrecommended. A modal that shows the warning while sends a police report in the backend would be more suitable for this.

**Pull request closed without merging**